Add log drains: list, add and remove endpoints (#94)
Scalingo/developers-documentation: _scalingo_api/log_drains.md

---

> title: Log drains

Reviewer: ```suggestion
title: Log Drains
```

Title in English have upper case for (almost) all words: https://www.notion.so/scalingooriginal/Writing-Documentation-at-Scalingo-bb7985d6ea16455894417e36ede09cea#3ff5d385e97e45bda37bc80cd618c725

layout: default
---

> # Log drains

Reviewer: ```suggestion
# Log Drains
```

> Log from an application can be redirect to a management service.

Reviewer: ```suggestion
Logs from an application can be redirected to a log management service.
```

> You can add multiple log drains from a single application.

Reviewer: In this sentence you talk about "log drains" but you never explained what a "log drain" is. You should add a sentence to explain.

```suggestion
You can add multiple log drains to a single application.
```

> You can also remove or list log drains by application.

Reviewer: ```suggestion
You can also remove or list log drains of an application.
```

> You will find how to enable this service from the CLI on the <a href="https://doc.scalingo.com/platform/app/log-drain">public documentation</a>

Reviewer: ```suggestion
You will find how to enable this service using the CLI on the <a href="https://doc.scalingo.com/platform/app/log-drain">public documentation</a>.
```

--- row ---

**Supported protocol and logs management services with their parameter**

{:.table}

> | Logs management service   | type        | token | host | port | drain_region | url | will format urls:                                    |

Reviewer: I think the "will format urls" column is useless. Customers don't care, it's kinda just an internal representation.

| Syslog                    | syslog      |       | ✓    | ✓    |              |     | tcp+tls://`HOST`:`PORT`                              |
| Self-hosted ELK stack     | elk         |       |      |      |              | ✓   | `URL`                                                |

--- row ---

**log drain attributes**

{:.table}
| field               | type    | description                                                      |
| ------------------- | ------  | ---------------------------------------------------------------- |
| type                | string  | Communication protocol with the logs management service          |
| token               | string  | Token used by logs management service for authentication         |
| host                | string  | Host of logs management service                                  |
| port                | string  | Port of logs management service                                  |
| drain-region        | string  | Region used by logs management service to identify their servers |
| url                 | string  | URL of self hosted ELK                                           |

> Note: `type` is mandatory.

Reviewer: Maybe add to the note that the mandatory attributes depend on the type?

||| col |||

--- row ---

> ## List log drains of an app

Reviewer: ```suggestion
## List Log Drains of an App
```

--- row ---

`GET https://$SCALINGO_API_URL/v1/apps/:app/log_drains`

||| col |||

Example request

```shell
curl -H 'Accept: application/json' -H 'Content-Type: application/json' -u ":$AUTH_TOKEN" \
  -X GET https://$SCALINGO_API_URL/v1/apps/example-app/log_drains
```

Returns 200 OK

```json
[
    {
        "app_id": "5af97be7ff688d0001228ffb",
        "url": "ovh://:id@tag1.logs.ovh.com:6514"
    },
    {
        "app_id": "5af97be7ff688d0001228ffb",
        "url": "syslog://:id@custom.logs.com:port"
    }
]
```

--- row ---

> ## Create a new log drain

Reviewer: ```suggestion
## Add a New Log Drain
```

--- row ---

`POST https://$SCALINGO_API_URL/v1/apps/:app/log_drains`

### Parameters

* `type`: can be any supported log management services type (e.g. papertrail, datadog...)

> * `token`: used by certain vendor for authentication

Reviewer: ```suggestion
* `token`: used by certain vendor for authentication (optional)
```

Can you add `(optional)` to the end of each optional items?

Author: I don't know if we need to add `optional` for the url parameter, because it is required in case of self hosted ELK as explained. What do you think ?

Reviewer: But `url` is not present in case of other types of log drains, is it? That's why I would make it optional. I.e. don't include `url` for all types.

Author: Okay it was just to be sure :+1:

* `host`: host of logs management service
* `port`: port of logs management service
* `drain_region`: region used by logs management service to identify their servers (e.g. region provided by datadog)
* `url`: URL of self hosted ELK

||| col |||

> Example request

Reviewer: ```suggestion
Example requests
```

```

--- row ---

> ## Delete a log drain

Reviewer: English title:

```suggestion
## Delete a Log Drain
```

--- row ---

`DELETE https://$SCALINGO_API_URL/v1/apps/:app/log_drains`

### Parameter

> * `url`: formatted URL return by create request or list request

Reviewer: I find this sentence hard to understand.
```suggestion
* `url`: logs drain URL to delete. You can get it by listing the logs drain of an application.
```
